src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Parser.cs

```diff
@@ -545,7 +545,9 @@ private bool IsAssignableTo(ITypeSymbol source, ITypeSymbol dest)
                 return conversion.IsReference && conversion.IsImplicit;
             }
 
-            private bool IsUnsupportedType(ITypeSymbol type)
+            private static HashSet<ITypeSymbol>? s_visitedTypes;
```

Comment:
Why is it static?  Can it be instance state and not require interlocked?

Reply:
I have changed that to be instance field instead.